tests/integration/wordpress_client.py

```diff
@@ -0,0 +1,122 @@
+import re
+import json
+
+import requests
+
+
+class WordpressClient:
+    """A very simple WordPress client for test purpose only"""
+
+    def __init__(self, host: str, username: str, password: str, is_admin: bool):
+        self.host = host
+        self.username = username
+        self.password = password
+        self._session = requests.session()
+        self.timeout = 10
+        self._login()
+        try:
+            self._session.get(
+                f"http://{self.host}/wp-json/",
+                timeout=self.timeout
+            ).json()
+        except requests.exceptions.JSONDecodeError:
+            if not is_admin:
+                raise ValueError("set options-permalink manually or login with an admin account")
+            self._set_options_permalink()
+
+    def _login(self):
```

Comment:
Do you think docstrings would be worthwhile here? Same comment applies for the rest of the functions in this module

Reply:
Fixed, thanks!